camera_cards

```diff
@@ -312,13 +321,45 @@ if [[ "${TAR}" == "YES" ]] ; then
     ## tar supplied input directory
     echo
     _report -g "Creating tar archive..."
+    _writelog "Tar archive process process started" "$(date +%FT%T)"
     tar -czvf "${AIPDIR}/objects/${MEDIAID}.tar.gz" -C "${CAMERA_CARD_DIR}" .
     _writelog "TAR ARCHIVE" "${MEDIAID}.tar.gz"
+    _writelog "Tar archive process process finished" "$(date +%FT%T)"
+fi
+
+# error reporting
+_report -b "Checking for errors during the camera_cards process..."
+if [[ -z $(cat "${TREE}") ]] || [[ -z $(cat "${MEDIAINFO_OUTPUT}") ]] || [[ -z $(cat "${EXIFTOOL_OUTPUT}") ]] || [[ -z $(cat "${FFPROBE_OUTPUT}") ]] ; then
+    _report -r "Metadata generation process created one or more blank files. Input files may be empty. Review metadata output at ${METADATA_OUTPUT_DIR}."
+    _writelog "POSSIBLE_ERROR_REVIEW" "Metadata generation process created one or more blank files"
+else
+    _report -g "Metadata generation process looks ok"
+fi
+if [[ "${FFMPEG_EXIT_CODE}" -gt 0 ]] || [[ "${FFMPEG_ERRORS}" -gt 0 ]] ; then
+    _report -r "FFmpeg may have encountered errors - review terminal output for more detail."
+else
+    _report -g "FFmpeg exit codes look ok"
+fi
+if [[ -s "${CONCATENATED_AUDIO_FILE}" ]] ; then
+    if [[ "${VIDEO_DURATION}" -ne "${AUDIO_DURATION}" ]] ; then
+    _report -r "Concatenated audio and video files were not the same length - review final merged file for sync at ${CONCATENATED_FILE} (video duration = ${VIDEO_DURATION}, audio duration = ${AUDIO_DURATION})"
+    _writelog "POSSIBLE_ERROR_REVIEW" "Concatenated audio and video files were not the same length"
+    else
+        _report -g "Concatenated audio and video process looks ok"
+    fi
+fi
+if [[ "${AIP}" == "YES" ]] ; then # check for losslessness of concatenated file by comparing bitstream md5s
+    OUTPUT_STREAM_MD5="$(_maketemp)"
+    ffmpeg -y -loglevel quiet -report -i "${CONCATENATED_VIDEO_FILE}" -f md5 -map 0:a? -map 0:v -c copy "${OUTPUT_STREAM_MD5}"
+    if [[ $(cat "${STREAM_MD5}") != $(cat "${OUTPUT_STREAM_MD5}") ]] ; then
+        _report -r "Concatenation process was not lossless! Review final merged file at ${CONCATENATED_VIDEO_FILE}"
```

Comment:
streamhash could do this better, as this test would require the audio and video to be interleaved in the same order, which isn't required for lossless remuxing. Like `V1,A1,A2,V2,A3,A4` and `V1,A1,A2,A3,V2,A4` are lossless but in different interleavement patterns, streamhash comparisons would separate to one hash per stream.

Reply:
Heya Dave, thanks! I agree that streamhash is better for this. Would you mind taking a look at this commit and letting me know if it makes sense: bcc5315f2a962870bb7d226c54eb611fddf56c45